src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -558,24 +565,50 @@ public static void approveSubmission(
             LOGGER.warn("Approving submissions did not affect any submissions. Something probably went wrong.");
         }
 
-        // Determine approvedScore
-        if (approvedScore == null) {
-            float bestScoreForPhase = submissionDao.getBestScoreForPhase(studentNetId, phase);
-            if (bestScoreForPhase < 0.0f) {
-                throw new RuntimeException("Cannot determine best score for phase without any submissions in the phase.");
+        if (submissionToUse == null) {
+            submissionToUse = submissionDao.getBestSubmissionForPhase(studentNetId, phase);
+            if (submissionToUse == null) {
+                throw new RuntimeException("No submission was provided nor found for phase " + phase + " with user " + studentNetId);
             }
-            approvedScore = SubmissionHelper.prepareModifiedScore(bestScoreForPhase, scoreVerification);
         }
-        if (approvedScore <= 0.0f) {
```

Comment:
What do you think about also validating that the approved score is a positive number?